src/jobs/ega/services/permissions.ts

```diff
@@ -183,12 +271,17 @@ export const processPermissionsForDataset = async (
       logger.error(
         `GET permissions for dataset ${datasetAccessionId} failed - ${permissions.message}`,
       );
-      // stop paging results if request completely fails to prevent endless loop
-      // can a retry mechanism be added here, if error is retryable?
-      paging = false;
+      permissionsRevocationResult.errors.push({
+        processName: 'getPermissionsForDataset',
+        status: permissions.status,
+        message: permissions.message,
+        datasetId: datasetAccessionId,
+      });
+      // TODO: add a max number of retries to prevent endless loop, then set paging = false?
```

Comment:
Calling out my comment here - with the request retry setup for 504 and connection reset errors in the EgaClient, i'm wondering if any "max retries" should be added there instead? So far in testing I haven't run into a scenario where pagination never stops 🤔

Reply:
Yes, we should probably setup a mechanism to prevent infinite loops on your retry mechanisms. You can give them a default max retries and then also accept a parameter to control from here how many max retries to allow.